src/covid_model_seiir_pipeline/pipeline/forecasting/model/ode_forecast.py

```diff
@@ -33,16 +35,176 @@
     )
 
 
+##############################
+# ODE parameter construction #
+##############################
+
+def build_model_parameters(indices: Indices,
+                           beta_params: Dict[str, float],
+                           thetas: pd.Series,
+                           covariates: pd.DataFrame,
+                           coefficients: pd.DataFrame,
+                           beta_scales: pd.DataFrame,
+                           vaccine_data: pd.DataFrame,
+                           scenario_spec: 'ScenarioSpecification') -> ModelParameters:
+    alpha = pd.Series(beta_params['alpha'], index=indices.full, name='alpha')
+    sigma = pd.Series(beta_params['sigma'], index=indices.full, name='sigma')
+    gamma1 = pd.Series(beta_params['gamma1'], index=indices.full, name='gamma1')
+    gamma2 = pd.Series(beta_params['gamma2'], index=indices.full, name='gamma2')
+
+    beta, beta_b117, beta_b1351 = forecast_beta(covariates, coefficients, beta_scales)
+
+    thetas = thetas.reindex(indices.full, level='location_id')
+
+    if ((1 < thetas) | thetas < -1).any():
+        raise ValueError('Theta must be between -1 and 1.')
+    if (sigma - thetas >= 1).any():
+        raise ValueError('Sigma - theta must be smaller than 1')
+
+    theta_plus = np.maximum(thetas, 0)
+    theta_minus = -np.minimum(thetas, 0)
+
+    vaccine_data = vaccine_data.reindex(indices.full, fill_value=0)
+    adjusted_vaccinations = adjust_vaccinations_for_variants(vaccine_data, covariates)
+
+    probability_cross_immune = pd.Series(scenario_spec.probability_cross_immune,
+                                         index=indices.full, name='probability_cross_immune')
+
+    return ModelParameters(
+        alpha=alpha,
+        beta=beta,
+        sigma=sigma,
+        gamma1=gamma1,
+        gamma2=gamma2,
+        theta_plus=theta_plus,
+        theta_minus=theta_minus,
+        **adjusted_vaccinations,
+        beta_b117=beta_b117,
+        beta_b1351=beta_b1351,
+        b117_prevalence=covariates['variant_prevalence_B117'],
+        b1351_prevalence=covariates['variant_prevalence_B1351'],
+        probability_cross_immune=probability_cross_immune,
+    )
+
+
+def adjust_vaccinations_for_variants(vaccine_data: pd.DataFrame, covariates: pd.DataFrame):
+    vaccine_data = vaccine_data.reindex(covariates.index, fill_value=0)
+    if 'variant_prevalence_B1351' in covariates.columns:
+        b1351_prevalence = covariates['variant_prevalence_B1351']
+    else:
+        b1351_prevalence = pd.Series(0, index=vaccine_data.index)
```

Comment:
Is this something you'd want to log/warn for? If this branch ever happened by accident it'd probably save you diagnostic/debug time...